test/Core.Test/OrganizationFeatures/AuthorizationHandlers/CollectionAccessAuthorizationHandlerBaseTests.cs

```diff
@@ -0,0 +1,228 @@
+    public async Task CanManageCollectionAccessAsync_MissingManageCollectionPermission_Failure(
+        SutProvider<CollectionUserAuthorizationHandler> sutProvider,
+        ICollection<Collection> collections,
+        ICollection<CollectionUser> collectionUsers,
+        ICollection<CollectionDetails> collectionDetails,
+        ICollection<CurrentContentOrganization> organizations)
+    {
+        var actingUserId = Guid.NewGuid();
+
+        // Simulate not having manage collection permission
+        foreach (var collectionDetail in collectionDetails)
+        {
+            collectionDetail.Manage = false;
+        }
+
+        // Ensure the user is not an owner/admin and does not have edit any collection permission
+        foreach (var org in organizations)
+        {
+            org.Type = OrganizationUserType.User;
+            org.Permissions.EditAnyCollection = false;
+        }
+
+        var context = new AuthorizationHandlerContext(
+            new[] { CollectionUserOperation.Create },
+            new ClaimsPrincipal(),
+            collectionUsers
+        );
+
+        sutProvider.GetDependency<ICurrentContext>().UserId.Returns(actingUserId);
+        sutProvider.GetDependency<ICurrentContext>().OrganizationMembershipAsync(Arg.Any<IOrganizationUserRepository>(), actingUserId).Returns(organizations);
+        sutProvider.GetDependency<ICollectionRepository>().GetManyByManyIdsAsync(Arg.Any<IEnumerable<Guid>>()).Returns(collections);
+        sutProvider.GetDependency<ICollectionRepository>().GetManyByUserIdAsync(actingUserId).Returns(collectionDetails);
+
+        await sutProvider.Sut.HandleAsync(context);
+        Assert.True(context.HasFailed);
+        await sutProvider.GetDependency<ICurrentContext>().ReceivedWithAnyArgs().OrganizationMembershipAsync(default, default);
+        await sutProvider.GetDependency<ICollectionRepository>().ReceivedWithAnyArgs().GetManyByManyIdsAsync(default);
+        await sutProvider.GetDependency<ICollectionRepository>().ReceivedWithAnyArgs()
+            .GetManyByUserIdAsync(default);
+    }
+
+    [Theory, BitAutoData, CollectionCustomization]
+    public async Task CollectionUserAuthorizationHandler_CollectionIdMap_Success(
+        SutProvider<CollectionUserAuthorizationHandler> sutProvider,
+        ICollection<CollectionUser> collectionUsers)
+    {
+        var actingUserId = Guid.NewGuid();
+        var context = new AuthorizationHandlerContext(
+            new[] { CollectionUserOperation.Create, CollectionUserOperation.Delete },
+            new ClaimsPrincipal(),
+            collectionUsers
+        );
+        var expectedCollectionIds = collectionUsers.Select(cu => cu.CollectionId);
+
+        sutProvider.GetDependency<ICurrentContext>().UserId.Returns(actingUserId);
+
+        await sutProvider.Sut.HandleAsync(context);
+
+        await sutProvider.GetDependency<ICollectionRepository>()
+            .Received(2) // Called twice, once for each operation requirement
```

Comment:
Hm, it only just clicked that all our db calls etc. are going to be repeated for each `OperationRequirement`. And often if we're modifying associations, we're both adding and deleting them, which means we'll often do this work twice when it's not really required. _And_ we'll do it for users and groups. (I assume - you might have some other insight here if you've been looking at the controller logic.)

Do you have any ideas on how to mitigate this? Maybe we do need to make the handlers and/or the operations a little less granular? (other suggestions welcome)